RELEASE/scripts/autoscend/iotms/mr2019.ash

```diff
@@ -939,4 +939,83 @@ void auto_deliberate_pizza()
-}
\ No newline at end of file
+}
+
+boolean changeSnapperPhylum(phylum toChange) {
+
+	if (!canChangeToFamiliar($familiar[Red-Nosed Snapper]) || toChange == $phylum[none]) {
+		return false;
+	}
+	string phylumString = (toChange == $phylum[mer-kin] ? "merkin" : toChange.to_string());
+	set_property("auto_snapperPhylum", phylumString);
+	return true;
+}
+
+boolean snapperPreAdventure(location loc) {
+	if (my_familiar() != $familiar[Red-Nosed Snapper]) {
+		return false;
+	}
+	
+	string desiredPhylum = get_property("auto_snapperPhylum");
+	if (desiredPhylum != "merkin" && desiredPhylum != "" && desiredPhylum.to_phylum() == $phylum[none]) {
+		auto_log_warning(`auto_snapperPhylum was set to bad value: {desiredPhylum}. Should be a valid phylum.`, "red");
+		remove_property("auto_snapperPhylum");
+		return false;
+	}
+
+	if (get_property("redSnapperPhylum") == desiredPhylum) {
+		auto_log_debug(`Red-Nosed Snapper is already guiding you towards {desiredPhylum}`);
+		return false;
+	}
+
+	// this is mainly in case autoChooseFamiliar switches to the Snapper due to no "better" +item familiars being available
+	// It is preferred that you do not rely on this to change phylum in a quest, call changeSnapperPhylum in the quest handling code instead.
+	if (desiredPhylum == "" && get_property("redSnapperProgress").to_int() < 7) {
+		switch (loc) {
+			case $location[The Penultimate Fantasy Airship]:
+			case $location[The Hidden Park]:
+			case $location[The Hidden Hospital]:
+			case $location[The Hidden Office Building]:
+			case $location[The Hidden Apartment Building]:
+			case $location[The Hidden Bowling Alley]:
+			case $location[The Copperhead Club]:
+			case $location[A Mob of Zeppelin Protesters]:
+			case $location[The Red Zeppelin]:
+			case $location[Inside the Palindome]:
+			case $location[The Neverending Party]:
+			case $location[South of The Border]:
+			case $location[The Valley of Rof L'm Fao]:
+				desiredPhylum = $phylum[dude].to_string(); // human musk (banisher)
+				break;
+			case $location[The Hole in the Sky]:
+				desiredPhylum = $phylum[constellation].to_string(); // micronova (yellow ray)
+				break;
+			case $location[The Smut Orc Logging Camp]:
+				desiredPhylum = $phylum[orc].to_string(); // boot flask (size 3 awesome booze)
+				break;
+			case $location[The Outskirts of Cobb's Knob]:
+			case $location[Cobb's Knob Barracks]:
+			case $location[Cobb's Knob Kitchens]:
+			case $location[Cobb's Knob Harem]:
+			case $location[Cobb's Knob Treasury]:
+			case $location[Cobb's Knob Laboratory]:
+				desiredPhylum = $phylum[goblin].to_string(); // guffin (size 3 awesome food)
+				break;
+			case $location[The "Fun" House]:
+				desiredPhylum = $phylum[horror].to_string(); // powdered madness (free kill)
+				break;
+			case $location[Twin Peak]:
+				// this is actually a dude heavy zone *but* we want to fight the topiary monsters for rusty hedge trimmers.
+				desiredPhylum = $phylum[beast].to_string();
+				break;
```

Comment:
Why not just make this a place where we always try to use the snapper? Seems like a reasonable way to get more trimmers. Unless there's a generally much better familiar to use there that's not coming to mind at the moment.

Reply:
The problem with the trimmers is that they're 15% drops. Cat Burglar would likely be a better thing to run there so we can heist them (I don't know if we do that yet though).

I'll have to see what we currently do in that zone and it's likely a Hardcore only issue as we should pull hedge trimmers in Normal (which I know we don't do yet, also on my list of stuff to fix).

Reply:
hedge trimmers only save about 2 adv per pull. we should probably not waste pulls on them.

Reply:
> hedge trimmers only save about 2 adv per pull. we should probably not waste pulls on them.

That depends on your resources. If you can only hit -10 combat rate from the 2 perms the turn savings are significantly higher.